src/tests/system/tests/test_passkey.py

```diff
@@ -241,3 +241,180 @@ def test_passkey__su_srv_not_resolvable(
+    :steps:
+        1. Check su authentication of the user.
+        2. Make server offline (by blocking traffic to the provider).
+        3. Bring SSSD offline explicitly.
+        4. Check su offline authentication of the user.
+    :expectedresults:
+        1. User su authenticated successfully.
+        2. Firewall rule added, traffic is dropped.
+        3. SSSD is offline.
+        4. Offline su authentication is successful.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    client.sssd.start()
+
+    # First time check authentication to cache the user
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+    # Render the provider offline
+    client.firewall.outbound.reject_destination(provider)
+
+    # There might be active connections that are not terminated by creating firewall rule.
+    # We need to terminated it by bringing SSSD to offline state explicitly.
+    client.sssd.bring_offline()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__user_fetch_from_cache(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Fetch a user from cache for LDAP, IPA, AD and Samba server
+    :setup:
+        1. Add a user in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check a user lookup.
+        2. Check a user from cache using ldbsearch command.
+    :expectedresults:
+        1. A user looked up successfully.
+        2. Successfully get the user from ldbsearch command.
+    :customerscenario: False
+    """
+
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    result = client.tools.id("user1")
+    output = client.ldb.search(path="/var/lib/sss/db/cache_test.ldb", basedn="cn=test,cn=sysdb")
+    assert result is not None
+    assert output["name=user1@test,cn=users,cn=test,cn=sysdb"] is not None
+    assert 'name=user1@test,cn=users,cn=test,cn=sysdb' in output.keys(), "user not find in cache"
```

Comment:
Would it be better to add a check for the passkey mapping in the cache also, SYSDB_USER_PASSKEY

Reply:
Checked the `userPasskey` attribute in list.